ci: add multiple clusters support

ldevulder opened the pull request:
Will fix https://github.com/rancher/elemental/issues/745.

Verification runs:
- [CLI-K3s-Multi_Cluster-RM_Stable](https://github.com/rancher/elemental/actions/runs/6694160766/job/18186841681?pr=1007)

Regression runs:
- [UI-K3s-RM_Stable](https://github.com/rancher/elemental/actions/runs/6695175400)
- [UI-RKE2-RM_Stable](https://github.com/rancher/elemental/actions/runs/6693672442)
- [CLI-K3s-RM_Stable](https://github.com/rancher/elemental/actions/runs/6693655110)
- [CLI-RKE2-RM_Stable](https://github.com/rancher/elemental/actions/runs/6695177519)

ldevulder committed:
ci: create SetMachineInventoryLabel function

Signed-off-by: Loic Devulder <ldevulder@suse.com>

ldevulder committed:
ci: add multi-cluster test

Signed-off-by: Loic Devulder <ldevulder@suse.com>

juadk commented on .github/workflows/cli-k3s-multi_cluster-rm_stable.yaml:
```diff
@@ -0,0 +1,41 @@
+      os_to_test: stable
+      # !! DEBUG !!
+      runner_template: elemental-e2e-ci-runner-spot-x86-64-template-n2-highmem-64-v4
+      test_type: multi
```
As we use this variable to difference `ui` and `cli` tests, could you use `multi_cli` instead?

ldevulder replied:
Yes, I split to `single_cli` and `multi_cli` (in addition to the already existing value `ui`).

ldevulder committed:
ci/cli: use different test_type for CLI tests

Use value `single_cli` for regulart CLI tests and `multi_cli` for
multi-cluster tests.

Signed-off-by: Loic Devulder <ldevulder@suse.com>

ldevulder committed:
ci/cli: use Canal as CNI for RKE2

Signed-off-by: Loic Devulder <ldevulder@suse.com>

juadk commented on .github/workflows/cli-k3s-multi_cluster-rm_stable.yaml:
```diff
@@ -0,0 +1,31 @@
+      iso_boot: true
+      k8s_version_to_provision: v1.26.8+k3s1
+      runner_template: elemental-e2e-ci-runner-spot-x86-64-template-n2-highmem-64-v4
+      test_type: multi
```
Should be `multi_cli` I think

ldevulder replied:
Yes, fixed, thanks!

juadk commented on .github/workflows/cli-rke2-multi_cluster-rm_stable.yaml:
```diff
@@ -0,0 +1,33 @@
+      iso_boot: true
+      k8s_version_to_provision: v1.26.8+rke2r1
+      runner_template: elemental-e2e-ci-runner-spot-x86-64-template-n2-highmem-64-v4
+      test_type: multi
```
Should be `multi_cli` I think

ldevulder replied:
Yes, fixed, thanks!

juadk reviewed (approved):
LGTM
